lib/httpcheck.js

```diff
@@ -192,15 +192,18 @@ var HttpChecker = {
 			}
 
 			if ( 0 === waiting_for ) {
-				if ( suicideSignal ) {
-//					HttpChecker.sendStats();
-					process.exit( SUICIDE_SIGNAL );
-				} else if ( process.memoryUsage().rss > maxMemUsage ) {
-//					HttpChecker.sendStats();
-					process.exit( EXIT_MAXRAMUSAGE );
-				} else if ( totalChecks > maxChecks ) {
-//					HttpChecker.sendStats();
-					process.exit( EXIT_MAXCHECKS );
+				// No outstanding callbacks and not available for work. Time to die!
+				if ( false === availableForWork ) {
```

Comment:
This is on purpose. Essentially if for any reason the worker is marker as NOT `availableForWork` and its work is complete then exit.
We could further improve it by adding an explicit check for memory limit but keep the final `else` with a new exit code.
This would require modifying `jetmon.js` a too (including the corresponding stats we send) but under the scope of this PR I chose to keep the changes minimal.

Reply:
I like this change. Can you clean it up to remove the leading whitespace in front of the comments and to combine the `0 === waiting_for` and the `false === availableForWork` checks into a single conditional?

Reply:
> Can you clean it up to remove the leading whitespace in front of the comments
Clean up ✅ 

> and to combine the 0 === waiting_for and the false === availableForWork checks into a single conditional
I believe it's best not to, because if `0 === waiting_for` and `true === availableForWork` , aka no outstanding request and the worker is still available for work (no exit) then it needs to reset `arrCheck` and set `running` to `false`.